src/main/scala/viper/silver/plugin/standard/inline/InlinePredicatePlugin.scala

```diff
@@ -28,9 +28,11 @@ class InlinePredicatePlugin extends SilverPlugin with ParserPluginTemplate
 
   override def beforeVerify(input: Program): Program = {
     val allPredIds = input.predicates.collect{ case p if p.body.nonEmpty => p.name }.toSet
-    val recursivePredIds = (checkRecursive(allPredIds, input) ++ checkMutualRecursive(allPredIds, input)).map(_.name)
-    val nonrecursivePredIds = allPredIds.diff(recursivePredIds)
-    val cond = { pred: String => nonrecursivePredIds(pred) }
+    val recursivePreds = checkRecursive(allPredIds, input) ++ checkMutualRecursive(allPredIds, input)
+    val recursivePredIds = recursivePreds.map(_.name)
+    val predsIdsCalledByOtherPreds = allPredIds.map(input.findPredicate).flatMap(nonRecursivePredsCalledBy).flatten
```

Comment:
`allPredIds` is just the names of every predicate, you can use `input.predicates` instead of `allPredIds.map(input.findPredicate)`

Reply:
I need more sleep

Reply:
naptime :sleeping_bed: :zzz:

Reply:
Hold on, not necessarily, `allPredIds` is the name of every predicate without an empty body - I knew there was a reason why I didn't do this. 

That said, I'll change it so that we don't keep coverting bt. ids and predicates.

Reply:
`nonRecursivePredsCalledBy` is doing a map into the body anyway, so even if the body is empty you'll just end up with an empty collection

Reply:
Actually shouldn't it specifically be `recursivePredIds` instead of `allPredIds`? We only care about the predicates that recursive predicates refer to
(In that case you'll have to do the `findPredicate` thing you're currently doing

Reply:
If we call it only on the recursive preds, I think we'll end up failing this test case
```
field f: Int

predicate F(this: Ref) {
  G(this)
}

predicate G(this: Ref) {
  acc(this.f)
}

predicate loop(this: Ref) {
  loop(this) && F(this)
}

method nestedPred(this: Ref)
  requires loop(this)
{
  unfold loop(this)
  unfold F(this)
  unfold G(this)
  this.f := 0
}
```

Reply:
You'll have to traverse through all subpredicates then
Any predicate that's indirectly used by a recursive predicate also gets excluded
Otherwise even in the absence of recursive predicates, any predicate that gets used by another one gets excluded, which we don't want